lisp/mastodon-tl.el

```diff
@@ -304,23 +373,17 @@ also render the html"
 
 (defun mastodon-tl--content (toot)
   "Retrieve text content from TOOT."
-  (let ((content (mastodon-tl--field 'content toot))
-        (shr-use-fonts nil))
-    (propertize (with-temp-buffer
-                  (insert content)
-                  (shr-render-region (point-min) (point-max))
-                  (buffer-string))
-                'face 'default)))
+  (let ((content (mastodon-tl--field 'content toot)))
+    (mastodon-tl--render-text content)))
 
 (defun mastodon-tl--toot (toot)
   "Display TOOT content and byline."
   (insert
    (concat
     (mastodon-tl--spoiler toot)
-    ;; remove two trailing newlines
-    (substring (mastodon-tl--content toot) 0 -2)
+    ;; remove trailing whitespace
+    (replace-regexp-in-string "[\t\n ]*\\'" "" (mastodon-tl--content toot))
     (mastodon-tl--media toot)
-    "\n\n"
```

Comment:
This cleans things up nicely, however it breaks navigation via `mastodon-tl--goto-next-toot`, which relies on replacing "\n\n\n | " to override the text-properties on the leading portion of the byline. This was always a bit of a hack and eventually needed to be replaced.

We could rework `mastodon-tl--byline` such that "\n | " is outside of the outermost `propertize`. This would essentially do the same thing as  `(while (search-forward "\n\n\n | " nil t) (replace-match "\n | "))` in `mastodon-tl--timeline`.

Reply:
Agreed.  Let's do this better.